From 6edb33a62c76a4193584b7a1ed436d54b43f1a91 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 13 Aug 2026 13:57:45 +0300
Subject: [PATCH 53/53] board: solidrun: lx2160acex7: select correct secure
 job-ring

optee-os for lx2160ardb target reserves caam job-ring 2 for lx2160ardb
builds, while u-boot deleted job-ring 3 in fdt fixup, effectively hiding
the wrong job-ring from OS.

This leads to one missing job-ring, and one that fails probe.

Fix the number for SolidRun LX2160 based platforms.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 configs/lx2160acex7_tfa_SECURE_BOOT_defconfig | 1 +
 configs/lx2160acex7_tfa_defconfig             | 1 +
 2 files changed, 2 insertions(+)

diff --git a/configs/lx2160acex7_tfa_SECURE_BOOT_defconfig b/configs/lx2160acex7_tfa_SECURE_BOOT_defconfig
index b35dff2bb6f..141952fb3d3 100644
--- a/configs/lx2160acex7_tfa_SECURE_BOOT_defconfig
+++ b/configs/lx2160acex7_tfa_SECURE_BOOT_defconfig
@@ -13,6 +13,7 @@ CONFIG_DEFAULT_DEVICE_TREE="fsl-lx2160a-cex7"
 CONFIG_FSPI_AHB_EN_4BYTE=y
 CONFIG_ARMV8_SEC_FIRMWARE_SUPPORT=y
 CONFIG_SEC_FIRMWARE_ARMV8_PSCI=y
+CONFIG_SEC_FIRMWARE_SEL1_JR_OFFSET=0x30000
 CONFIG_AHCI=y
 CONFIG_OF_BOARD_FIXUP=y
 CONFIG_REMAKE_ELF=y
diff --git a/configs/lx2160acex7_tfa_defconfig b/configs/lx2160acex7_tfa_defconfig
index 160b25f979c..c1f39b6c5f7 100644
--- a/configs/lx2160acex7_tfa_defconfig
+++ b/configs/lx2160acex7_tfa_defconfig
@@ -14,6 +14,7 @@ CONFIG_FSPI_AHB_EN_4BYTE=y
 CONFIG_SYS_MONITOR_LEN=958464
 CONFIG_ARMV8_SEC_FIRMWARE_SUPPORT=y
 CONFIG_SEC_FIRMWARE_ARMV8_PSCI=y
+CONFIG_SEC_FIRMWARE_SEL1_JR_OFFSET=0x30000
 CONFIG_ENV_ADDR=0x20500000
 CONFIG_PCI=y
 CONFIG_AHCI=y
-- 
2.43.0

